source/extensions/filters/http/tap/tap_config_impl.h

```diff
@@ -45,6 +48,7 @@ class HttpPerRequestTapperImpl : public HttpPerRequestTapper, Logger::Loggable<L
   HttpTapConfigImplSharedPtr config_;
   const uint64_t stream_id_;
   std::vector<bool> statuses_;
+  std::shared_ptr<envoy::data::tap::v2alpha::BufferedTraceWrapper> trace_;
```

Comment:
Why shared?

Reply:
Because we need to pass a shared pointer to the submit trace function (because it may get captured to a lambda). I will add a comment.